services/src/main/java/org/fao/geonet/api/GlobalExceptionController.java

```diff
@@ -138,7 +170,7 @@ public ApiError webappExceptionHandler(final Exception exception) {
         GeoPublisherException.class,
         NoResultsFoundException.class})
     public ApiError NotFoundHandler(final Exception exception) {
-        return new ApiError("not_found", exception.getClass().getSimpleName(), exception.getMessage());
+        return new ApiError("not_found", exception);
```

Comment:
Should not be call `updateExceptionLocale(exception, request);` in all the methods handling at least GeoNetwork custom exceptions?

Reply:
@josegar74 - yes I think we should if we have the request object to get the language.  In this case NotFoundHandler does not have the `HttpServletRequest request` object as a function parameter.  
I noticed that some functions have the `HttpServletRequest request` and some functions don't.  I was not sure if I would simply add them without causing issues. I believe it can be added later as a separate PR. 
The purpose of this PR was mostly to introduce the LocallizedException classes and give a couple samples to see if the concept is acceptable by the community.  Once approved then we can start updating and testing other usages.